File: 백준/Silver/28278. 스택 2/스택 2.java
import java.util.*;
import java.io.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;
    static int N;

    public static void main(String[] args) throws Exception {
        Stack<Integer> stack = new Stack<>();
        N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            int command = Integer.parseInt(st.nextToken());

            switch (command) {
                case 1:
                    stack.push(Integer.parseInt(st.nextToken()));
                    break;
                case 2:
                    if (stack.isEmpty())
                        bw.write("-1\n");
                    else
                        bw.write(stack.pop() + "\n");
                    break;
                case 3:
                    bw.write(stack.size() + "\n");
                    break;
                case 4:
                    if (stack.isEmpty())
                        bw.write("1\n");
                    else
                        bw.write("0\n");
                    break;
                case 5:
                    if (stack.isEmpty())
                        bw.write("-1\n");
                    else
                        bw.write(stack.peek() + "\n");
                    break;
            }
        }
        bw.close();
    }
}

File: 백준/Silver/28278. 스택 2/스택 2.py
import sys


def input(): return sys.stdin.readline().rstrip()


N = int(input())

stack = []
for _ in range(N):
    command = input().split()
    if command[0] == '1':
        stack.append(command[1])
    elif command[0] == '2':
        print(stack.pop() if stack else '-1')
    elif command[0] == '3':
        print(len(stack))
    elif command[0] == '4':
        print(int(not stack))
    elif command[0] == '5':
        print(stack[-1] if stack else -1)